pgmpy/inference/ExactInference.py

```diff
@@ -32,20 +32,21 @@ def _get_working_factors(self, evidence):
         -------
         dict: Modified working factors.
         """
+
         working_factors = {
-            node: {factor for factor in self.factors[node]} for node in self.factors
+            node: {(factor, None) for factor in self.factors[node]} for node in self.factors
         }
 
         # Dealing with evidence. Reducing factors over it before VE is run.
         if evidence:
             for evidence_var in evidence:
                 for factor in working_factors[evidence_var]:
```

Comment:
Same here. Using `factor[0]` makes the code less readable as it is difficult to deduce why we are only using the first element and what are the other elements. Using `for factor, _ in ...` makes it explicit that there are only 2 elements and the second element is not used anywhere in the loop.

Reply:
You are right, but I don't understand what you propose for this specific line of code (there's no factor[0] here).

Reply:
Sorry, I meant to comment on the next line with `factor[0].reduce`. Changing this loop will avoid the indexing there.